ImageGallery/src/org/sleuthkit/autopsy/imagegallery/datamodel/DrawableDB.java

```diff
@@ -688,6 +704,65 @@ private void insertOrUpdateFile(DrawableFile f, @Nonnull DrawableTransaction tr,
         }
     }
 
+
+   /**
+     * Gets all known data source object ids from data_sources table
+     *
+     * @return list of known data source object ids
+     */
+    public Set<Long> getKnownDataSourceIds() throws TskCoreException {
+        Statement statement = null;
+        ResultSet rs = null;
+        Set<Long> ret = new HashSet<>();
+        dbReadLock();
+        try {
+            statement = con.createStatement();
+            rs = statement.executeQuery("SELECT ds_obj_id FROM datasources "); //NON-NLS
+            while (rs.next()) {
+                ret.add(rs.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new TskCoreException("SQLException thrown when calling 'DrawableDB.getKnownDataSourceIds()", e);
+        } finally {
+            if (rs != null) {
+                try {
+                    rs.close();
+                } catch (SQLException ex) {
+                    LOGGER.log(Level.SEVERE, "Error closing result set after executing  getKnownDataSourceIds", ex); //NON-NLS
+                }
+            }
+            if (statement != null) {
+                try {
+                    statement.close();
+                } catch (SQLException ex) {
+                    LOGGER.log(Level.SEVERE, "Error closing statement after executing  getKnownDataSourceIds", ex); //NON-NLS
+                }
+            }
+            dbReadUnlock();
+        }
+        return ret;
+    }
+    
+    
+    /**
+     * Insert given data source object id into data_sources table
+     * 
+     * @param dsObjectId data source object id to insert
+     */
+    public void  insertDataSource(long dsObjectId) {
```

Comment:
Perhaps indicate that this is insert if not exists. At least in the documentation.